app/vmalert/rule/alerting.go

```diff
@@ -36,13 +35,13 @@ type AlertingRule struct {
 
 	q datasource.Querier
 
-	alertsMu sync.RWMutex
+	AlertsMu sync.RWMutex
 	// stores list of active alerts
-	alerts map[uint64]*notifier.Alert
+	Alerts map[uint64]*notifier.Alert
 
 	// state stores recent state changes
 	// during evaluations
-	state *ruleState
+	State *ruleState
```

Comment:
Having public field for private struct seems weird. It also exposes its embedded mutex methods.

Reply:
The same goes for NewRuleState method and similar

Reply:
Fixed, thanks